jmix-ui/ui/src/main/java/io/jmix/ui/screen/Screen.java

```diff
@@ -120,6 +120,7 @@ protected ScreenData getScreenData() {
 
     protected void setScreenData(ScreenData data) {
         this.screenData = data;
+        this.screenData.setScreenId(getId());
```

Comment:
ScreenData is also used in ScreenFragment.
Move this Screen-ScreenData linkage to UiControllerUtils as it done for ViewControllerUtils.
Check if it works ok for fragments.